src/ape/types/address.py

```diff
@@ -1,13 +1,38 @@
-from typing import Union
+from importlib import import_module
+from typing import Annotated, Any, Optional, Union
 
-from eth_typing import ChecksumAddress as AddressType
-from ethpm_types import HexBytes
+from eth_pydantic_types import Address as _Address
+from eth_pydantic_types import HashBytes20, HashStr20
+from eth_typing import ChecksumAddress
+from pydantic_core.core_schema import ValidationInfo
 
-RawAddress = Union[str, int, HexBytes]
+RawAddress = Union[str, int, HashStr20, HashBytes20]
 """
 A raw data-type representation of an address.
 """
 
+
+class _AddressValidator(_Address):
+    """
+    An address in Ape. This types works the same as
+    ``eth_pydantic_types.address.Address`` for most cases,
+    (validated size and checksumming), unless your ecosystem
+    has a different address type, either in bytes-length or
+    checksumming algorithm.
+    """
+
+    @classmethod
+    def __eth_pydantic_validate__(cls, value: Any, info: Optional[ValidationInfo] = None) -> str:
+        return (
+            getattr(import_module("ape"), "convert")(value, AddressType)
```

Comment:
this can't use ManagerAccessMixin?

Reply:
Turns out it can! Plus this `getattr => import_module` is really gross, not sure what i was thinking.

But, first I have to fix an issue.. turns out optionals are not working right with our new pydantic types. We need to tell pydantic about the optional somewhere it seems...

Reply:
done